test/upgrade/olm_upgrade_test.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	. "github.com/project-codeflare/codeflare-common/support"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/project-codeflare/codeflare-operator/test/e2e"
+)
+
+var (
+	namespaceName  = "test-ns-olmupgrade"
+	appwrapperName = "mnist"
+	jobName        = "mnist-job"
+)
+
+func TestMNISTCreateAppWrapper(t *testing.T) {
+	test := With(t)
+	test.T().Parallel()
+
+	// Create a namespace
+	namespace := CreateTestNamespaceWithName(test, namespaceName)
+
+	// Delete namespace only if test failed
+	defer func() {
+		if t.Failed() {
+			DeleteTestNamespace(test, namespace)
+		} else {
+			StoreNamespaceLogs(test, namespace)
+		}
+	}()
```

Comment:
@Srihari1192 @sutaakar out of curiosity, why not using the "standard" way, where test support does that automatically?

Reply:
@astefanutti In this Upgrade context, we are using the same namespace in after operator upgrade test.  As **NewTestNamespace** will delete the namespace by default after test complete , so we added this supported methods in [codeflare-common ](https://github.com/project-codeflare/codeflare-common/blob/988ba1da9069e1d8919da11761227daee6bce552/support/namespace.go#L57)

Reply:
> It may be better to implement the upgrade as part of the Go test

This will couple the test with specific upgrade strategy (using OLM, overriding existing deployment with new oneliner, upgrade using ODH). Personally I would prefer keep test implementation aside from deployment/upgrade, to keep the test reusable for any strategy.

Reply:
It makes sense. Sounds good 👍🏼.